crates/router/src/connector/stripe.rs

```diff
@@ -1861,7 +1861,11 @@ impl api::IncomingWebhook for Stripe {
                 match details.event_data.event_object.metadata {
                     // if order_id is present
                     Some(meta_data) => api_models::webhooks::ObjectReferenceId::PaymentId(
-                        api_models::payments::PaymentIdType::PaymentAttemptId(meta_data.order_id),
+                        api_models::payments::PaymentIdType::PaymentAttemptId(
```

Comment:
Since our condition is based on order_id and not metadata, match statement can be rewritten as 

```suggestion
match details
                    .event_data
                    .event_object
                    .metadata
                    .and_then(|metadata| metadata.order_id)
                {
                    // if order_id is present
                    Some(order_id) => api_models::webhooks::ObjectReferenceId::PaymentId(
                        api_models::payments::PaymentIdType::PaymentAttemptId(order_id),
                    ),
                    // else used connector_transaction_id
                    None => api_models::webhooks::ObjectReferenceId::PaymentId(
                        api_models::payments::PaymentIdType::ConnectorTransactionId(
                            details.event_data.event_object.id,
                        ),
                    ),
                }
```